test/admin-hpa/run-local-tests.sh

```diff
@@ -0,0 +1,83 @@
+#!/bin/sh
+#set -x
+
+# Run the tests locally in parallel
+JOB_COUNT=6
+PARALLEL_FEATURE_FILE=tags-parallel.txt
+SERIES_FEATURE_FILE=tags-serial.txt
+LOG_DIR=out
+BIN=parallel
+
+# Ensure we have a parallel binary
+if ! type -p ${BIN} >/dev/null 2>/dev/null; then
+  echo "${BIN} not found"
+  echo ""
+  echo "You can install it using homebrew:"
+  echo '   `brew install parallel`'
+  echo ""
+  exit 1
+fi
+
+clean ()
+{
+  rm -rf $LOG_DIR
+  mkdir $LOG_DIR
+}
+
+mybanner ()
+{
+  echo "+------------------------------------------+"
+  printf "| %-40s |\n" "`date`"
+  echo "|                                          |"
+  printf "|`tput bold` %-40s `tput sgr0`|\n" "$@"
+  echo "+------------------------------------------+"
+}
+
+# print to the console in RED
+writeError ()
+{
+  MSG=$1
+  echo -e "\033[31m ${MSG} \033[0m"
+}
+
+# print to the console in YELLOW
+writeInfo ()
+{
+  MSG=$1
+  echo -e "\033[33m ${MSG} \033[0m"
+}
+
+writeSuccess ()
+{
+    MSG=$1
+  echo -e "\033[92m ${MSG} \033[0m"
+}
+
+test ()
+{
+  FILE=$1
```

Comment:
can you document these params at the top of the file please, and also check they exist?

Reply:
😀. This is the documentation.

In a bash function $1 refers to the 1st parameter passed to the function, $2 to the 2nd and so on.   So while it's possible to just use $1 or $2 within the function body, this assignment at least makes it readable.

Reply:
yes, i am aware of the parameter indexing and $0 being the script itself.  

If you check for the existence of any required params near the top of the file and inform the user if anything is missing, it explains itself and stops it running without anything missing.

Here is [an example](https://github.com/DFEAGILEDEVOPS/MTC/blob/master/deploy/keys/function-masterkey-update-keyvault.sh)

Reply:
It's no dealbreaker, so if you're averse to such an addition just leave as is...

Reply:
i've now realised after discussion that these are local function arguments, not script arguments.
resolving this...